solidity/test/integration/utils/ecdsa-wallet-registry.ts

```diff
@@ -0,0 +1,269 @@
+// TODO: Utils in this file are pulled from @keep-network/ecdsa test utils.
+// We should consider exposing them in @keep-network/ecdsa for an external usage.
+
+/* eslint-disable no-await-in-loop */
+import { deployments, ethers, helpers } from "hardhat"
+
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import type {
+  ContractTransaction,
+  BytesLike,
+  BigNumber,
+  BigNumberish,
+  Contract,
+  Signer,
+} from "ethers"
+import type { WalletRegistry, SortitionPool } from "../../../typechain"
+
+// Number of members in the ECDSA Wallet.
+const WALLET_SIZE = 100
+
+export async function registerOperator(
+  walletRegistry: WalletRegistry,
+  sortitionPool: SortitionPool,
+  stakingProvider: Signer,
+  operator: Signer
+): Promise<number> {
+  await walletRegistry
+    .connect(stakingProvider)
+    .registerOperator(await operator.getAddress())
+
+  await walletRegistry.connect(operator).joinSortitionPool()
+
+  const operatorID = await sortitionPool.getOperatorID(
+    await operator.getAddress()
+  )
+
+  return operatorID
+}
+
+export async function produceEcdsaDkgResult(
+  walletRegistry: WalletRegistry,
+  groupPublicKey: BytesLike,
+  relayEntry: BigNumberish,
+  startBlock: number
+): Promise<{ approveDkgResultTx: ContractTransaction }> {
+  const seed = calculateDkgSeed(relayEntry, startBlock)
+
+  const {
+    dkgResult,
+    submitter,
+    submitDkgResultTx: dkgResultSubmissionTx,
+  } = await signAndSubmitDkgResult(
+    walletRegistry,
+    groupPublicKey,
+    seed,
+    startBlock
+  )
+
+  await helpers.time.mineBlocksTo(
+    dkgResultSubmissionTx.blockNumber +
+      (
+        await walletRegistry.dkgParameters()
+      ).resultChallengePeriodLength.toNumber()
+  )
+
+  const approveDkgResultTx = await walletRegistry
+    .connect(submitter)
+    .approveDkgResult(dkgResult)
+
+  return { approveDkgResultTx }
+}
+
+export async function updateWalletRegistryDkgResultChallengePeriodLength(
+  walletRegistry: WalletRegistry,
+  governance: Signer,
+  dkgResultChallengePeriodLength: BigNumberish
+): Promise<void> {
+  const walletRegistryGovernance = await ethers.getContractAt(
+    (
+      await deployments.getArtifact("WalletRegistryGovernance")
+    ).abi,
+    await walletRegistry.governance()
+  )
+
+  await walletRegistryGovernance
+    .connect(governance)
+    .beginDkgResultChallengePeriodLengthUpdate(dkgResultChallengePeriodLength)
+
+  await helpers.time.increaseTime(
+    await walletRegistryGovernance.governanceDelay()
+  )
+
+  await walletRegistryGovernance
+    .connect(governance)
+    .finalizeDkgResultChallengePeriodLengthUpdate()
+}
+
+async function selectGroup(
+  sortitionPool: Contract,
+  seed: BigNumber
+): Promise<Operator[]> {
+  const identifiers = await sortitionPool.selectGroup(
+    WALLET_SIZE,
+    ethers.utils.hexZeroPad(seed.toHexString(), 32)
+  )
```

Comment:
Why don't we use [`selectGroup` from `WalletRegistry`](https://github.com/keep-network/keep-core/blob/9cf3620dbd8ccd016dc193b2eb23f0a2509dc87c/solidity/ecdsa/contracts/WalletRegistry.sol#L1150-L1152)? This is what the client will call, it does not need to bother about the `SortitionPool`.

Also, if we use this function I think we should be able to get rid of `calculateDkgSeed` and all the complexity calculating and passing the seed introduces.

Reply:
Nice catch! Yeah, I missed this function. I copied the code from ecdsa's test utils.

Reply:
c8f7d9b60cdd121f41bcac32bf2a2495b668bdad